### PR TITLE
fix(PolicyRulesTab): RHICOMPL-2042 - Fix dedicated action for legacy rules tab

### PR DESCRIPTION
**Pull request by bastilian**
This fixes issues the policy details rules tab when the policy has no system and setting the minor OS version failed.

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: src/SmartComponents/PolicyDetails/PolicyRulesTab.js [hunk 1]
```diff
@@ -12,44 +12,48 @@ import * as Columns from '@/PresentationalComponents/RulesTable/Columns';
...
-        },
-      ]}
-      options={{
-        dedicatedAction: <EditRulesButtonToolbarItem policy={policy} />,
-      }}
-    />
-  </React.Fragment>
-);
+const PolicyRulesTab = ({ loading, policy }) => {
+  const DedicatedAction = () => <EditRulesButtonToolbarItem policy={policy} />;
```
Comment (vkrizan) [inline src/SmartComponents/PolicyDetails/PolicyRulesTab.js:16]: `useDedicatedAction` might be changed to accept component instances/elements which would avoid creating these wrappers.

For example:
```
React.isValidElement(DedicatedActionOption) 
  ? DedicatedActionOption
  : <DedicatedActionOption {...(selected && { selected })} />
```
from https://github.com/RedHatInsights/compliance-frontend/pull/1322/files#diff-27e743131e4a1c6a482a583448b8354d6bbd9f4acb109ccccb8bc7235f76cbb7R9

**Review by vkrizan — APPROVED** [review]
Works well.

File: src/SmartComponents/PolicyDetails/PolicyRulesTab.js (hunk 1 above)
Comment (bastilian) [reply src/SmartComponents/PolicyDetails/PolicyRulesTab.js:16]: @vkrizan What would be the advantage of that?

Comment (vkrizan) [reply src/SmartComponents/PolicyDetails/PolicyRulesTab.js:16]: As stated, no wrapper would be necessary. It is more common to pass existing instance, such as action button, to a dedicated action without needing to read selected items. In summary, less code, less instances and less calls.

**Pull request merged**